### PR TITLE
JOINDIN-579 - API for retrieving list of talk types (categories)

### PR DESCRIPTION
**Pull request by dstockto**
GET .../v2.1/categories
GET .../v2.1/categories/<ID>

### CODE REVIEW

File: src/config/routes/2.1.json [hunk 1]
```diff
@@ -174,6 +174,20 @@
     },
 
     {
+        "path": "/categories$",
```
Comment (chrishoult) [inline src/config/routes/2.1.json]: Most of the other routes have `/?` at the end to cope with a trailing slash or not; please add this in.

Comment (dstockto) [inline src/config/routes/2.1.json]: Will do.

**Comment by dstockto** [comment]
@chrishoult Updated to allow for optional trailing slash

File: src/controllers/DefaultController.php [hunk 2]
```diff
@@ -8,6 +8,7 @@ public function handle(Request $request, $db)
 
         // just add the available methods, with links
         $retval['events']          = $request->base . '/' . $request->version . '/events';
+        $retval['categories']      = $request->base . '/' . $request->version . '/categories';
```
Comment (lornajane) [inline src/controllers/DefaultController.php]: I'm not sure how useful it is to offer this link at the entry level of the API, I mostly think of this as a list of places a user wants to go

Comment (dstockto) [inline src/controllers/DefaultController.php]: I'm happy to remove the entry and resubmit the PR if you'd like. On the other hand, do we have a place for a developer to go who wants to integrate w/ joind.in and needs to know what APIs are available?

**Comment by lornajane** [comment]
We do have docs, which should include this, but that's probably the only place it would be

**Comment by dstockto** [comment]
I'll remove the entry and resubmit then. Thank you

**Comment by akrabat** [comment]
The `/languages' endpoint has `name` and this one has `title` for the same piece of data. Should we be consistent and make `/categories` also use `name`?

**Comment by lornajane** [comment]
Merged but if you wanted to write a test to go with it, that would also be fine ;)

**Pull request merged**